tools/core-build-tasks/src/tasks/copy.ts

```diff
@@ -0,0 +1,147 @@
+ *
+ * These should be a scheme for validation purposes
+ */
+export type ConfigurationFormat = {
+  [outputPath: string]: FilesToCopy | FilesToCopy[];
+};
+
+const COPY_CONFIG_DEFAULT = "./config/copyTask.config.json";
+
+export type CopyTaskParameters = {
+  packageRoot: string;
+  /**
+   * if not specified, defaults to config/copyTask.config.json under current package folder
+   */
+  copyTaskConfigPath?: string;
+};
+
+/**
+ * A just task which copies files to a specified output location. The configuration passed in uses the following format:
+ * {
+ *   [outputPath]: string[]
+ * }
+ *
+ * Where there may be multiple output paths, and for each output path there may be multiple files. The Output Path accepts
+ * defined constants:
+ *
+ *   PROJECT_NAME
+ *
+ * These constants are replaced at task execution with a value provided by the process environment.
+ *
+ * Files to copy must be specified relative to the location of the configuration file. Only files may be specified. For files, the
+ * following constants will be replaced:
+ *
+ *   PACKAGE_VERSION
+ *
+ */
+export function copyTask(params: CopyTaskParameters) {
+  return () => {
+    const configPath =
+      params.copyTaskConfigPath ?? Path.resolve(__dirname, COPY_CONFIG_DEFAULT);
+    const config = JsonFile.load(configPath) as ConfigurationFormat | undefined;
+
+    if (!config) {
+      throw new Error(
+        "Unable to process copy configuration, please review configuration file."
+      );
+    }
+
+    const projectName = getOrThrowFromProcess("PROJECT_NAME");
+
+    // Load package.json for current package
+    const packageJson = JsonFile.load(
+      Path.resolve(params.packageRoot, "./package.json")
+    ) as { version: string };
+    const packageVersion = packageJson["version"];
+    if (!packageVersion) {
+      throw new Error(
+        "Unable to get current package version. Make sure to configure package root correctly."
+      );
+    }
+
+    let deploymentPath: string | undefined = undefined;
+    try {
+      const product =
+        getOrThrowFromProcess<MinecraftProduct>("MINECRAFT_PRODUCT");
+      deploymentPath = getGameDeploymentRootPaths()[product];
+    } catch (_) {
+      throw new Error(
+        "Unable to get deployment path. Make sure to configure package root correctly."
+      );
+    }
+
+    if (deploymentPath === undefined) {
+      throw new Error(
+        "Deployment path is undefined. Make sure to configure package root correctly."
+      );
+    }
+
+    // For each output path, replace tokens with env values
+    for (const outputDirectoryRaw of Object.keys(config)) {
+      const rawFiles = config[outputDirectoryRaw];
+      const filesToCopy = Array.isArray(rawFiles) ? rawFiles : [rawFiles];
+      for (const fileToCopyConfig of filesToCopy) {
+        // Note: order of replacement matters to avoid replacing partial matches initially
```

Comment:
Not a relevant comment here since the only replaced string is PROJECT_NAME

Reply:
Removed